include/hdlConvertor/hdlObjects/hdlTypeDec.h

```diff
@@ -0,0 +1,69 @@
+#pragma once
+#include <vector>
+#include <utility>
+#include <memory>
+#include <hdlConvertor/hdlObjects/named.h>
+#include <hdlConvertor/hdlObjects/iHdlObj.h>
+
+
+namespace hdlConvertor {
+namespace hdlObjects {
+
+class iHdlExpr;
+class HdlVariableDef;
+
+/*
+ * HDL AST node for module declaration
+ * (part with ports and params for Verilog, Entity for VHDL)
+ * */
+class HdlTypeDec: public WithNameAndDoc, public iHdlObj {
+public:
+	// enum to indicate which class of type declaration this is, useful
+	// when visiting the AST
+	enum TypeClass {hdltc_error, hdltc_incomplete, hdltc_enum, hdltc_array, 
+		hdltc_struct, hdltc_class, hdltc_physical, hdltc_integral, 
+		hdltc_float, hdltc_access, hdltc_file, hdltc_protected};
+
+	HdlTypeDec(const std::string& name);
+	virtual TypeClass typeClass() {return hdltc_error;};
+	virtual ~HdlTypeDec();
+};
+
+class HdlEnumTypeDec: public HdlTypeDec {
+public:
+	// VHDL character enumerations are stored as strings including the
+	// single-quotes, e.g. "'U'"s; SV allows identifiers to have an
+	// associated value, while VHDL scalar types do not have this concept
+	std::vector<std::pair<std::string, std::unique_ptr<iHdlExpr>>> ids;
+
+	TypeClass typeClass() override {return hdltc_enum;};
+	virtual ~HdlEnumTypeDec();
+};
+
+class HdlArrayTypeDec: public HdlTypeDec {
+public:
+	std::vector<std::unique_ptr<iHdlExpr>> indexes;
```

Comment:
In order to support multi-dimensional arrays, my current array data type class has a vector of `iHdlExpr` for indexes.  However, the expression parser returns a single pointer to `iHdlExpr` when parsing an index constraint.

https://github.com/Nic30/hdlConvertor/blob/869ec94218a56133be8d56182297d34ca3bf3212/src/vhdlConvertor/exprParser.cpp#L224

I'll investigate changing the return of `VhdlExprParser::visitIndex_constraint` to a vector, hopefully on Friday (UTC-6).